mender-auth/ipc/client.cpp

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <mender-auth/ipc/client.hpp>
+
+#include <functional>
+#include <iostream>
+#include <string>
+
+#include <common/common.hpp>
+#include <common/error.hpp>
+#include <common/events_io.hpp>
+#include <common/http.hpp>
+#include <common/log.hpp>
+#include <common/path.hpp>
+
+namespace mender {
+namespace ipc {
+
+using namespace std;
+
+namespace path = mender::common::path;
+
+
+const IPCErrorCategoryClass IPCErrorCategory;
+
+const char *IPCErrorCategoryClass::name() const noexcept {
+	return "IPCErrorCategory";
+}
+
+string IPCErrorCategoryClass::message(int code) const {
+	switch (code) {
+	case NoError:
+		return "Success";
+	case SetupError:
+		return "Error during setup";
+	case RequestError:
+		return "HTTP client request error";
+	case ResponseError:
+		return "HTTP client response error";
+	case APIError:
+		return "API error";
+	case UnauthorizedError:
+		return "Unauthorized error";
+	default:
+		return "Unknown";
+	}
+}
+
+error::Error MakeError(IPCErrorCode code, const string &msg) {
+	return error::Error(error_condition(code, IPCErrorCategory), msg);
+}
+
+namespace type {
+namespace webserver {
```

Comment:
Should maybe the file also be called `webserver.cpp` instead of `client.cpp`?

Reply:
I understand the argument. My original thinking was that it could have multiple types, so that we have:

`client::type::<sometype>`

But mby ur right.

Reply:
Yes, that's good thinking, but we'd anyway want those to be in different files, no? I'm considering that later we may have even more types than webserver and dbus, it might be "zephyr_ipc", or something like that. So it makes sense to split them, IMHO.

Reply:
I called it `server.cpp` for now?

Reply:
👍